frontend/src/components/RecordsList.tsx

```diff
@@ -94,20 +98,38 @@ export default function RecordsList({
   const [rowCount, setRowCount] = useState<number>(0);
   const [uniqueSampleCount, setUniqueSampleCount] = useState<number>(0);
 
+  const grid = useRef<AgGridReactType>(null);
   const navigate = useNavigate();
 
   // note that we aren't using initial fetch
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [initialFetch, { loading, error, data, fetchMore, refetch }] =
     lazyRecordsQuery({
       variables: {
-        options: { limit: 20, offset: 0 },
+        options: {
+          limit: 20,
+          offset: 0,
+          sort: defaultSort,
+        },
         ...lazyRecordsQueryAddlVariables,
       },
     });
 
   const totalCountNodeName = `${dataName}Connection`;
 
+  const getSortModel = useCallback(() => {
+    const sortModel = grid.current?.columnApi
+      ?.getColumnState()
+      .filter(({ sort }) => sort)
+      .map(({ colId, sort }) => ({
+        [colId]: sort?.toUpperCase() as SortDirection,
+      }));
+    if (sortModel && sortModel.length > 0) {
+      return sortModel;
+    }
+    return defaultSort;
+  }, [defaultSort]);
```

Comment:
This function gets called whenever the frontend queries. It helps the frontend pass along the sort setting to the GraphQL custom resolver. If the user manually sort some column, use that. Otherwise, use the default sort setting (e.g. `importDate` for Requests view).